src/hex.rs

```diff
@@ -219,6 +219,51 @@ impl ToHex for [u8] {
     }
 }
 
+/// A struct implementing [`io::Write`] that converts what's written into it
+/// in an hex String
+///
+/// If you already have the data to be converted in a `Vec<u8>` use [`ToHex`]
+/// but if you have an encodable object, by using this you avoid the
+/// serialization to `Vec<u8>` by going directly to `String`.
+///
+/// Note that to achieve better perfomance than [`ToHex`] the struct must be
+/// created with the right `capacity` of the end hex string result so that the
+/// inner `String` doesn't re-allocate
+#[cfg(any(test, feature = "std", feature = "alloc"))]
+#[cfg_attr(docsrs, doc(cfg(any(test, feature = "std", feature = "alloc"))))]
+pub struct HexWriter(String);
+
+#[cfg(any(test, feature = "std", feature = "alloc"))]
+#[cfg_attr(docsrs, doc(cfg(any(test, feature = "std", feature = "alloc"))))]
+impl HexWriter {
+    /// Creates a new [`HexWriter`] specify the `capacity` of the inner `String` that will contain
+    /// the hex value
+    pub fn new(capacity: usize) -> Self {
```

Comment:
I think `new` should take no parameters and we should have a separate `with_capacity` constructor which takes a capacity.

Reply:
What mentioned @tcharding is the reason why I avoided having the `new` without parameters. @apoelstra do you think it's okay?